static/js/formatters-internal.js

```diff
@@ -348,22 +376,80 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
 
   const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
 
-  const dynamicUrl = atLeastAsLarge
+  return atLeastAsLarge
     ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
     : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
+}
 
-  return Object.assign(
-    {},
-    image,
-    {
-      url: dynamicUrl.replace('http://', 'https://')
+/**
+ * Given an EU image url, returns the dynamic url.
+ *
+ * @param {Object} image The image object. (e.g.
+ * {
+ *   url: 'https://a.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg',
+ *   sourceUrl: 'https://a.mktgcdn.com/p/UN9RPhz0V9D8bNZ3XfNpkGnAk6ikFhVmgvntlBjVyMA/1200x675.jpg',
+ *   width: 1200,
+ *   height: 675,
+ * })
+ * @param {string} desiredSize The desired size of the image ('<width>x<height>')
+ * @param {boolean} atLeastAsLarge Whether the image should be at least as large as the desired
+ *                                 size in one dimension or smaller than the desired size in both
+ *                                 dimensions.
+ * @returns {string} A dynamic url (e.g. 'https://dyn.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg/width=200,fit=contain')
+ */
+function _getEuImageDynamicUrl(image, desiredSize, atLeastAsLarge) {
+  let fullSizeWidth, fullSizeHeight;
+  if (image.width) {
+    fullSizeWidth = image.width;
+  }
+  if (image.height) {
+    fullSizeHeight = image.height;
+  }
+
+  if (image.sourceUrl && (!fullSizeWidth || !fullSizeHeight)) {
+    const [urlWithoutExtension, _] = _splitStringOnIndex(image.sourceUrl, image.sourceUrl.lastIndexOf('.'));
+    const [__, dimensions] = _splitStringOnIndex(urlWithoutExtension, urlWithoutExtension.lastIndexOf('/') + 1);
+    const fullSizeDims = dimensions.split('x');
+
+    fullSizeWidth = Number.parseInt(fullSizeDims[0]);
```

Comment:
if `image.width` is present (but not `image.height`), would we still want to replace the `fullSizeWidth` value?

Reply:
Yep. I have a test case to cover that ("when choosing the biggest image under threshold, use dimensions from the sourceUrl if width/height does not exist")

Reply:
that test case has a smaller width in the `sourceUrl` than in the `image.width`, so that makes sense to me. but, if the `image.width` were smaller than `sourceUrl`, would we want to use the `image.width` instead?

Reply:
My concern with using one dimension from image.width/image.height and the other dimension from sourceUrl is that the ratio might be off. We just use this as a fallback. So if the client wants to have a precise dimension, they can always specify it with the desiredDims

Reply:
I believe the dimensions in the `sourceUrl` would always match `image.width` and `image.height`. This is just to make sure we wouldn't get any errors from `image.width/height` not existing